demos/howtos/multiple_likelihoods.py

```diff
@@ -0,0 +1,180 @@
+                                                                         exactSolution=x_exact,
+                                                                         SNR=SNR)
+
+# Plot data, exact data and exact solution
+plt.figure()
+data1.plot(label='data')
+problemInfo1.exactData.plot(label='exact data')
+problemInfo1.exactSolution.plot(label='exact solution')
+plt.legend()
+
+# %%
+# Set up the second model
+# -----------------------
+
+observation_grid_map2 = None
+if set_up == "multi_observation":
+        # Observe on the second half of the domain
+	observation_grid_map2 = lambda x: x[np.where(x>=.5)]
+
+# The source term signal
+if set_up == "multi_source":
+	source2 = lambda xs: 20*np.sin(2*xs)+20.1
+else:
+	source2 = source1
+
+# Obtain the forward model from the test problem
+model2, data2, problemInfo2 = cuqi.testproblem.Poisson_1D.get_components(dim=dim,
+                                                                         endpoint=endpoint,
+                                                                         field_type=field_type,
+                                                                         field_params={
+                                                                             'n_steps': n_steps},
+                                                                         observation_grid_map=observation_grid_map2,
+                                                                         exactSolution=x_exact,
+                                                                         SNR=SNR)
+
+# Plot data, exact data and exact solution
+plt.figure()
+data2.plot(label='data2')
+problemInfo2.exactData.plot(label='exact data2')
+problemInfo2.exactSolution.plot(label='exact solution2')
+plt.legend()
+
+# %%
+# Create the prior
+# ----------------
+# Create the prior for the Bayesian parameter `x`, which is the conductivity (or diffusivity) of the medium. We use a Gaussian prior.
+
+x = cuqi.distribution.GaussianCov(np.zeros(
+	model1.domain_dim), 3*np.ones(model1.domain_dim), geometry=model1.domain_geometry)
+
+# %%
+# Create the likelihoods
+# ----------------------
+
+# Estimate the data noise standard deviation
+sigma_noise1 = np.linalg.norm(problemInfo1.exactData)/SNR
+sigma_noise2 = np.linalg.norm(problemInfo2.exactData)/SNR
+
+# Create the data distributions
+y1 = cuqi.distribution.Gaussian(mean=model1, std=sigma_noise1,
+                                corrmat=np.eye(model1.range_dim), geometry=model1.range_geometry)
+y2 = cuqi.distribution.Gaussian(mean=model2, std=sigma_noise2,
+                                corrmat=np.eye(model2.range_dim), geometry=model2.range_geometry)
+
+# %%
+# Create the posterior (multiple likelihoods)
+# -------------------------------------------
+
+z_joint = cuqi.distribution.JointDistribution(x,y1,y2)(y1=data1, y2=data2)._as_stacked() # _as_stacked() is needed to stack the random variables but it is a temporary hack that will be not needed in the future.
```

Comment:
Is there an issue to remove the need for _as_stacked?

Reply:
Issue #123 created